Bottom: 76610836e9ca6e4bec4cea68a7e2dbbb297e09e0
Top:    ef0b1cf59ba6d797a04099e4761e4ebbc9b796c6
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2012-12-02 15:18:31 +0000

Workaround crash where contour is passed as NULL to poly_PreContour














---

diff --git a/src/polygon.c b/src/polygon.c
index e15a65a..5ceaa98 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -307,6 +307,16 @@ original_poly (PolygonType * p)
       if (n == p->PointN - 1 ||
           (hole < p->HoleIndexN && n == p->HoleIndex[hole] - 1))
         {
+          if (contour == NULL)
+            {
+              printf ("How did that escape - did the loop iterate zero times??\n");
+              POLYGONPOINT_LOOP (p);
+                {
+                  printf ("Hello\n");
+                }
+              END_LOOP;
+              return NULL;
+            }
           poly_PreContour (contour, TRUE);
 
           /* make sure it is a positive contour (outer) or negative (hole) */
